eagleproject/core/blockchain/harvest/transaction_history.py

```diff
@@ -755,6 +758,114 @@ def process_transaction(transaction):
 
     return analyzed_transactions
 
+def ensure_analyzed_wrap_transactions(from_block, to_block, start_time, end_time, account='all'):
```

Comment:
Huge amount of code seems to overlap between this function and `ensure_analyzed_transactions`. I think it would make sense to create a new function that has an extra bool `is_ousd`:
```
__ensure_transaction_analysis(from_block, to_block, start_time, end_time, account='all', mode):
    is_ousd = mode === 'ousd'
    .... (rest of code)
```
and both functions can call the newly created function: 
```
def ensure_analyzed_transactions(from_block, to_block, start_time, end_time, account='all'):
    __ensure_transaction_analysis(from_block, to_block, start_time, end_time, account, 'ousd'

def ensure_analyzed_wrap_transactions(from_block, to_block, start_time, end_time, account='all'):
    __ensure_transaction_analysis(from_block, to_block, start_time, end_time, account, 'wousd'
```

This way you can reduce the duplicated code and increase maintainability -> There is less overall code to maintain and when you make a change in one of the functions you don't need to keep in mind that you need to make change in the other one that has duplicated code as well